extensions/wikia/PortableInfobox/tests/nodes/NodeDataTest.php

```diff
@@ -39,93 +39,409 @@ public function sourceDataProvider() {
 				'<format>my {{{test2}}}$$$</format>' .
 				'<default>{{#switch: {{{test3|}}}|{{{test4}}}|{{{test5|kdjk|sajdkfj|}}}]] }}</default>' .
 				'</data>',
 				[ 'test1' => 'blabla' ],
 				[
-					'test1' => 'next label (test1)',
-					'test2' => 'next label (test2)',
-					'test3' => 'next label (test3)',
-					'test4' => 'next label (test4)',
-					'test5' => 'next label (test5)'
+					'test1' => [
+						'label' => 'next label (test1)',
+						'primary' => true
+					],
+					'test2' => [
+						'label' => 'next label (test2)'
+					],
+					'test3' => [
+						'label' => 'next label (test3)'
+					],
+					'test4' => [
+						'label' => 'next label (test4)'
+					],
+					'test5' => [
+						'label' => 'next label (test5)'
+					]
 				]
 			],
-			[ '<data>' .
+			[
+				'<data>' .
 				'<label>last label</label>' .
 				'<format>my {{{test2}}}$$$</format>' .
 				'<default>{{#switch: {{{test3|}}}|{{{test4}}}|{{{test5|kdjk|sajdkfj|}}}]] }}</default>' .
 				'</data>',
-				[ ],
+				[],
+				[
+					'test2' => [
+						'label' => 'last label (test2)'
+					],
+					'test3' => [
+						'label' => 'last label (test3)'
+					],
+					'test4' => [
+						'label' => 'last label (test4)'
+					],
+					'test5' => [
+						'label' => 'last label (test5)'
+					]
+				]
+			]
+		];
+	}
+
+	/**
+	 * @covers       \Wikia\PortableInfobox\Parser\Nodes\Node::getMetadata
+	 * @dataProvider metadataDataProvider
+	 *
+	 * @param $markup
+	 * @param $params
+	 * @param $expected
+	 */
+	public function testMetadata( $markup, $params, $expected ) {
+		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
+
+		$this->assertEquals( $expected, $node->getMetadata() );
+	}
+
+	public function metadataDataProvider() {
+		return [
+			[
+				'<infobox><data source="test"></data></infobox>',
+				[],
+				[
+					[
+						'type' => 'data',
+						'sources' => [
+							'test' => [
+								'label' => '',
+								'primary' => true
+							]
+						]
+					]
+				]
+			],
+			[
+				'<infobox><image source="test"/></infobox>',
+				[],
+				[
+					[
+						'type' => 'image',
+						'sources' => [
+							'test' => [
+								'label' => '',
+								'primary' => true
+							]
+						]
+					]
+				]
+			],
+			[
+				'<infobox><title source="test"/></infobox>',
+				[],
+				[
+					[
+						'type' => 'title',
+						'sources' => [
+							'test' => [
+								'label' => '',
+								'primary' => true
+							]
+						]
+					]
+				]
+			],
+			[
+				'<infobox><data source="test"><default>{{{test}}}</default></data></infobox>',
+				[],
+				[
+					[
+						'type' => 'data',
+						'sources' => [
+							'test' => [
+								'label' => '',
+								'primary' => true
+							]
+						]
+					]
+				]
+			],
+			[
+				'<infobox><group>' .
+				'<image source="test1" />' .
+				'<title source="test2" />' .
+				'</group></infobox>',
+				[],
+				[
+					[
+						'type' => 'group',
+						'metadata' => [
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "metadata" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)